docs: Create a documentation for GCP deployment and write addition informations about AWS (#1334)
aneoconsulting/ArmoniK: .docs/content/1.installation/3.aws/2.aws-deployment-troubleshooting.md

# AWS Deployment Troubleshooting Guide 🛠️

This guide covers common issues you might encounter while deploying your application on AWS and provides solutions to address them.

## 1. Error: Partition Not Found

If you encounter a Partition Not Found error during deployment or runtime, ensure the following:

> - Set the Replication Factor to 1 in the [`parameters.tfvars`](https://github.com/aneoconsulting/ArmoniK/blob/main/infrastructure/quick-deploy/aws/parameters.tfvars):

Reviewer: ```suggestion
- Set the Replication Factor to 1 in the [`parameters.tfvars`](infrastructure/quick-deploy/aws/parameters.tfvars):
```
Can you do it for the other ones ? This points on the current branch whereas main

Author: Normally the use of the https://github is the correct way to do because using just infrastructure/quick-deploy is not working locally. We had to use this path : `../../../../infrastructure/quick-deploy `. Is it the right way to do or we can conserve the https link ?

```hcl
default = {
    node_selector = { service = "workers" }
    replicas = 1 # HERE
}
```

- Check Spelling in Configuration Files.
- Double-check for any typos in partition names or configurations. Even minor spelling mistakes can cause the error.

For example in [`parameters.tfvars`](https://github.com/aneoconsulting/ArmoniK/blob/main/infrastructure/quick-deploy/aws/parameters.tfvars):

```hcl
default = {
    node_selector = { service = "workers" }   
    replicas = 1 # HERE
```

instead of

```hcl
defautl = {
    node_selector = { service = "workers" }   
    replicas = 1 # HERE

```

## 2. Error: Incorrect URL or Missing Endpoint

> If your application fails to connect to a service, verify the following:

Reviewer 2: ```suggestion
If your application fails to connect to a service, you need to verify the following possibilities.
```

### Use the Correct URL Generated by AWS

> After deploying your application or service, AWS typically generates a URL for the *control_plane* or other services. Make sure to:

Reviewer 2: ```suggestion
After deploying your application or service, AWS typically generates an URL for the *control_plane* and the other services. Make sure to:
```

- Copy the URL provided by AWS during deployment.

> - Update the client configuration with the correct endpoint before building your project.

Reviewer 2: not sure to understand this sentence. The client is not started as an executable with command line option? The ip is written on hard and so need the client to be rebuild?

```bash
docker run --rm --name <container name> client --endpoint "http://<ip>:5001" --partition <partition name>
```

- Replace <ip> with the actual IP or hostname of the deployed service.
- Replace <partition_name> with the appropriate partition name you’re using.
- Always double-check that the endpoint and partition names are spelled correctly to avoid runtime errors.

## 3. Missing Images

If you encounter an error related to the Docker image not being found or incorrectly configured, follow these steps to ensure the image is built and referenced correctly.

### Ensure the Worker Image is Built Correctly

You need to build the Docker image for the worker using the following command:

```bash
docker build -f "./csharp/native/LinearSubTasking/Worker/Dockerfile" -t subtasking "./csharp/native/"
```

- **-f**: Specifies the path to the Dockerfile for your worker.
- **-t subtasking**: Tags the image as subtasking, which will be used as the image name.
- **"./csharp/native/"**: Specifies the context for the build (i.e., the root directory containing your source files).

Voici la documentation mise à jour pour inclure la gestion des erreurs liées aux images Docker :

> 3. Error: Docker Image Not Found

Reviewer 2: don't you need to provide a reference to the page?

Author: The whole page will be removed. Since it handles the case of troubleshooting for a particular case, the C# sample.
